scrapy_zyte_api/addon.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional
+
+from scrapy.settings import BaseSettings
+
+from scrapy_zyte_api import ScrapyZyteAPIDownloaderMiddleware
+
+
+class Addon:
+    @staticmethod
+    def _check_settings(settings: BaseSettings) -> Optional[str]:
+        if (
+            settings.getwithbase("DOWNLOAD_HANDLERS")["http"]
+            != "scrapy.core.downloader.handlers.http.HTTPDownloadHandler"
+        ):
+            return "'http' value in the 'DOWNLOAD_HANDLERS'"
+        if (
+            settings.getwithbase("DOWNLOAD_HANDLERS")["https"]
+            != "scrapy.core.downloader.handlers.http.HTTPDownloadHandler"
+        ):
+            return "'https' value in the 'DOWNLOAD_HANDLERS'"
+        if (
+            settings.get("REQUEST_FINGERPRINTER_CLASS")
+            != "scrapy.utils.request.RequestFingerprinter"
+        ):
+            return "'REQUEST_FINGERPRINTER_CLASS'"
+        return None
+
+    def update_settings(self, settings: BaseSettings) -> None:
```

Comment:
By the way, what do you think about configuring SCRAPY_POET_PROVIDERS as well?

Reply:
It's a bit tricky though, because it's not clear how to un-configure it if user doesn't want it.

Reply:
Yeah, I tried this and as there are no priorities in the dict and the addon code runs later than the settings from `settings.py` are loaded you can't disable it via `settings.py`.

Reply:
We could add an addon-specific setting which would tell the addon to not add the provider, not sure if this is a good way to do it. In any case it's a good example of what would addon authors want.

Reply:
If we provide a setting to disable the provider I wonder how often will it be used, because it one line of code in settings.py to enable the provider manually.

Reply:
Yes, it doesn't seem to worth it.